### PR TITLE
All meny close automatically. Test activity if no activity. Small bugs with text

### PR DESCRIPTION
**Pull request by Alona-Kirichenko**
- adding "react-native-outside-press"
- Now all many in app close automatically when a user presses on something else at a screen
- Changing:  Yesterday to Igår.  Today to Idag
- Changing: Lägga till bild to Lägg till bild
- Fixing bug. Now it won't crash if a request to get activity returns undefined 
- Now at HomePage, it will check connected activities. If user has an activity that not exist in db, it will delete this activity from the connected_activities array ayt user document.  If connected_activities has no connected activity, then it will connect #jagsomväntarpåakticitet activity.
         I added some roles at Firebase Firestore Database. 

           //Check if user logged in and want apdate it's own connected_activities
    function isUserWantToUpdateConnectedActivities(userId) {
    	return (request.auth != null &&
      request.auth.uid == userId && 
      request.data.diff(resource.data).affectedKeys().hasOnly(['connected_activities']))
    }
    
 ```tsx 
// Check that user wants to add only test activity if array connected_activities is empty
function isUserWantConnectTestActivity(userId) {
  return (
    request.auth != null &&
    request.auth.uid == userId && 
    resource.data.connected_activities.size() == 0 && 
    request.data.connected_activities.size() == 1 && 
    request.data.connected_activities[0] == 'cVkZbjsrXs6YKk3Qli0b'
  )
}
```

### CODE REVIEW

File: src/firebase-functions/get.js [hunk 1]
```diff
@@ -91,7 +91,8 @@ export const getActivityInformation = async (activityId) => {
 
     return Promise.resolve(activity);
   } catch (error) {
-    return Promise.reject(error);
+    console.log(Promise.reject(error));
```
Comment (Mattan1999) [inline src/firebase-functions/get.js:94]: Remove

File: src/firebase-functions/getTS/get.ts [hunk 2]
```diff
@@ -135,11 +133,37 @@ export const getUserByStatus = async (isActive: boolean = true) => {
...
 export const getActivityByID = async (
   activityID: Activity["id"],
-): Promise<Activity> => {
+): Promise<Activity | null> => {
   try {
     let data = await firestore().collection("Activities").doc(activityID).get();
     return Promise.resolve(activityObject(data));
   } catch (error) {
-    return Promise.reject(error);
+    console.log("getActivityByID: ", error);
```
Comment (Mattan1999) [inline src/firebase-functions/getTS/get.ts:141]: Remove

File: src/firebase-functions/getTS/get.ts [hunk 3]
```diff
@@ -135,11 +133,37 @@ export const getUserByStatus = async (isActive: boolean = true) => {
...
+        title: querySnapshot?.data()?.activity_title,
+        city: querySnapshot?.data()?.activity_city,
+        photo: querySnapshot?.data()?.activity_photo,
+        imageUrl: querySnapshot?.data()?.imageUrl,
+      };
+    }
+
+    return Promise.resolve(activity);
+  } catch (error) {
+    console.log(Promise.reject(error));
```
Comment (Mattan1999) [inline src/firebase-functions/getTS/get.ts:166]: remove

**Review by Mattan1999 — APPROVED** [review]
